assertion failed: private method dynamicPower in object X

## minimized code

```Scala
package playground

import scala.quoted._
import scala.quoted.matching._

object X {

  inline def power(n: Int, x: Double): Double =
    ${ powerImpl('n, 'x) }

  private def powerImpl(nExpr: Expr[Int], xExpr: Expr[Double])(given QuoteContext): Expr[Double] =
    nExpr match {
      case Const(n1) => '{ 42.0 }
      case _ => '{ dynamicPower($nExpr, $xExpr) }
    }

  private def dynamicPower(n: Int, x: Double): Double = {
    println(s"dynamic: $n^$x")
    if (n == 0) 1.0
    else if (n % 2 == 0) dynamicPower(n / 2, x * x)
    else x * dynamicPower(n - 1, x)
  }
}
```
create some worksheet:
```scala
import playground.X
X.power(f, 2)
```

## Compilation output

```scala
Exception in thread "main" java.lang.AssertionError: assertion failed: private method dynamicPower in object X in C:\Users\dmitrii.naumenko\Desktop\dev\dotty-example-project\src\main\scala\playground\X.scala accessed from constructor rs$line$10$ in rs$line$10
    at dotty.DottyPredef$.assertFail(DottyPredef.scala:17)
    at dotty.tools.dotc.transform.ExpandPrivate.ensurePrivateAccessible(ExpandPrivate.scala:84)
    at dotty.tools.dotc.transform.ExpandPrivate.transformIdent(ExpandPrivate.scala:92)
    at dotty.tools.dotc.transform.ExpandPrivate.transformIdent(ExpandPrivate.scala:91)
    at dotty.tools.dotc.transform.MegaPhase.goIdent(MegaPhase.scala:551)
    at dotty.tools.dotc.transform.MegaPhase.transformNamed$1(MegaPhase.scala:225)
    at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:394)
    at dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:270)
    at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:396)
    at dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:306)
    at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:396)
    at dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:287)
    at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:396)
    at dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:324)
    at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:396)
    at dotty.tools.dotc.transform.MegaPhase.transformStat$2(MegaPhase.scala:406)
    at dotty.tools.dotc.transform.MegaPhase.$anonfun$1(MegaPhase.scala:409)
    at scala.collection.immutable.List.mapConserve(List.scala:444)
    at dotty.tools.dotc.transform.MegaPhase.transformStats(MegaPhase.scala:409)
    at dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:286)
    at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:396)
    at dotty.tools.dotc.transform.MegaPhase.mapDefDef$1(MegaPhase.scala:245)
    at dotty.tools.dotc.transform.MegaPhase.transformNamed$1(MegaPhase.scala:248)
    at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:394)
    at dotty.tools.dotc.transform.MegaPhase.transformSpecificTree(MegaPhase.scala:400)
    at dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:336)
    at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:396)
    at dotty.tools.dotc.transform.MegaPhase.transformNamed$1(MegaPhase.scala:251)
    at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:394)
    at dotty.tools.dotc.transform.MegaPhase.transformStat$2(MegaPhase.scala:404)
    at dotty.tools.dotc.transform.MegaPhase.$anonfun$1(MegaPhase.scala:409)
    at scala.collection.immutable.List.mapConserve(List.scala:444)
    at dotty.tools.dotc.transform.MegaPhase.transformStats(MegaPhase.scala:409)
    at dotty.tools.dotc.transform.MegaPhase.mapPackage$1(MegaPhase.scala:356)
    at dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:359)
    at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:396)
    at dotty.tools.dotc.transform.MegaPhase.transformUnit(MegaPhase.scala:415)
    at dotty.tools.dotc.transform.MegaPhase.run(MegaPhase.scala:427)
    at dotty.tools.dotc.core.Phases$Phase.runOn$$anonfun$1(Phases.scala:315)
    at scala.collection.immutable.List.map(List.scala:219)
    at dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:316)
    at dotty.tools.dotc.Run.runPhases$4$$anonfun$4(Run.scala:161)
    at dotty.runtime.function.JProcedure1.apply(JProcedure1.java:15)
    at dotty.runtime.function.JProcedure1.apply(JProcedure1.java:10)
    at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
    at dotty.tools.dotc.Run.runPhases$5(Run.scala:171)
    at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:179)
    at dotty.runtime.function.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
    at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:65)
    at dotty.tools.dotc.Run.compileUnits(Run.scala:186)
    at dotty.tools.dotc.Run.compileUnits(Run.scala:128)
    at dotty.tools.repl.ReplCompiler.runCompilationUnit(ReplCompiler.scala:156)
    at dotty.tools.repl.ReplCompiler.compile(ReplCompiler.scala:166)
    at dotty.tools.repl.ReplDriver.compile(ReplDriver.scala:229)
    at dotty.tools.repl.ReplDriver.interpret(ReplDriver.scala:193)
    at dotty.tools.repl.ReplDriver.run$$anonfun$1(ReplDriver.scala:136)
    at dotty.tools.repl.ReplDriver.withRedirectedOutput(ReplDriver.scala:148)
    at dotty.tools.repl.ReplDriver.run(ReplDriver.scala:137)
    at dotty.tools.languageserver.worksheet.ReplProcess$.main(ReplProcess.scala:13)
    at dotty.tools.languageserver.worksheet.ReplProcess.main(ReplProcess.scala)
```

## expectation
`X.power` shouldn't compile cause it is inlined and uses private method

--- a/compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala
+++ b/compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala
@@ -66,8 +66,11 @@ trait QuotesAndSplices {
         typedTypeApply(untpd.TypeApply(untpd.ref(defn.QuotedTypeModule_of.termRef), tree.quoted :: Nil), pt)(using quoteContext).select(nme.apply).appliedTo(qctx)
       else
         typedApply(untpd.Apply(untpd.ref(defn.QuotedRuntime_exprQuote.termRef), tree.quoted), pt)(using pushQuotes(qctx)).select(nme.apply).appliedTo(qctx)
-    tree1.withSpan(tree.span)
+    makeInlineable(tree1.withSpan(tree.span))
   }
+
+  private def makeInlineable(tree: Tree)(using Context): Tree =
+    ctx.compilationUnit.inlineAccessors.makeInlineable(tree)
 
   /** Translate `${ t: Expr[T] }` into expression `t.splice` while tracking the quotation level in the context */
   def typedSplice(tree: untpd.Splice, pt: Type)(using Context): Tree = {

--- a/tests/neg-macros/i7068-c.scala
+++ b/tests/neg-macros/i7068-c.scala
@@ -2,7 +2,7 @@ def species(using quoted.Quotes) = '{
   case object Bar // error
   case class FooT() // error
   ${
-    case object Baz // ok
+    case object Baz // error
     ???
   }
   FooT()

--- a/tests/pos-macros/i12948/Macros_1.scala
+++ b/tests/pos-macros/i12948/Macros_1.scala
@@ -1,0 +1,9 @@
+package mylib
+import scala.quoted.*
+
+object Main:
+  protected def foo: Unit = {}
+  inline def fooCaller: Unit = foo
+  inline def fooCallerM: Unit = ${ fooMacro }
+  def fooMacro(using Quotes): Expr[Unit] =
+    '{ foo }

--- a/tests/pos-macros/i12948/Test_2.scala
+++ b/tests/pos-macros/i12948/Test_2.scala
@@ -1,0 +1,5 @@
+import mylib.Main
+
+object Test:
+  Main.fooCaller
+  Main.fooCallerM

--- a/tests/pos-macros/i12948b.scala
+++ b/tests/pos-macros/i12948b.scala
@@ -1,0 +1,8 @@
+import scala.quoted.*
+
+package foo {
+  trait Bar:
+    def bar(using Quotes) = '{ Baz }
+
+  private[foo] object Baz
+}

--- a/tests/pos-macros/i12948b.scala
+++ b/tests/pos-macros/i12948b.scala
@@ -1,8 +1,0 @@
-import scala.quoted.*
-
-package foo {
-  trait Bar:
-    def bar(using Quotes) = '{ Baz }
-
-  private[foo] object Baz
-}

--- a/tests/pos-macros/i8208/Macros_1.scala
+++ b/tests/pos-macros/i8208/Macros_1.scala
@@ -1,0 +1,22 @@
+package playground
+
+import scala.quoted._
+
+object X {
+
+  inline def power(n: Int, x: Double): Double =
+    ${ powerImpl('n, 'x) }
+
+  private def powerImpl(nExpr: Expr[Int], xExpr: Expr[Double])(using Quotes): Expr[Double] =
+    nExpr match {
+      case Expr(n1) => '{ 42.0 }
+      case _ => '{ dynamicPower($nExpr, $xExpr) }
+    }
+
+  private def dynamicPower(n: Int, x: Double): Double = {
+    println(s"dynamic: $n^$x")
+    if (n == 0) 1.0
+    else if (n % 2 == 0) dynamicPower(n / 2, x * x)
+    else x * dynamicPower(n - 1, x)
+  }
+}

--- a/tests/pos-macros/i8208/Test_2.scala
+++ b/tests/pos-macros/i8208/Test_2.scala
@@ -1,0 +1,2 @@
+import playground.X
+def test(x: Int) = X.power(x, 2)

